### PR TITLE
Qcon two

### PR DESCRIPTION
**Pull request by kevinhodges**
### Blog title
QConn New York 2015 key takeaways - Part 2

### Blog description
Second part (of 2) summarising my notes from this years QCon in New York.

#### Author
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 1
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 2
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

### CODE REVIEW

**Comment by JonathanBD** [comment]
Taking Editor 1

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+### Developing cultural intelligence
+#### Daniel Seltzer
+
+Kev: Good talk on something that is very hard to define, sorry not many notes, was listening intently.
+
+Culture happens in real-time, at the boundaries between people and groups.  When shit hits the fan, the culture IS the process that happens between people.
+#### Leaders set the culture.  You don't have to have control (i.e. be a CEO) but if you set the culture, people look to you as a leader.
+Example, if you live the culture not just act it and others see that they will follow or immitate.
+You can't force culture on to everybody.  Essentially he says, if your values conflict with a person, push then out, they are no good for your team or for you.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "Push them out" 
Aside: What a charming sentiment.

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Sharing informtion and openness about what you are doing is key.
+
+
+### Failure as a service
+#### Kolton Andrus (Chaos Engineer on the Netflix platform)
+
+Kev: Another Netflix guy, spoke well about the chaos monkey system they've built and how they've implemented it, sounds quite simple in principle...
+
+Failure testing helps makes systems immune to failure which in turn prevents larger outages and increase uptime/response cycles
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Full stops after bullets inconsistent.

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+Can be limited to apps, regions etc to reduce risks
+
+3. Where (injection points)
+points that fail already or look like they might fail
+
+4. Failure itself
+Each service/module implements a snippet to inject deplay where required.
+
+Demo of failure ui, turned it on, personalised stuff dropped off home page defaulting to a "fixed" list
+There's a "Halt all failure" button red top right to stop everything, feels really really risky.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "feels really, really risky..."

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+#### You cant build anti-fragile systems but you can build an anti-fragile culture.  Getting people to think about this stuff is the most important thing.
+ 
+Testing in production is sustainable and helps improve performance long term.
+
+
+### Clojurescript demo
+#### Jearvon Darrie
+
+Kev: I'm afraid this was the only take away of any value.  A good one though right?
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Can probably lose the "Kev:"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+Rules...
+
+1, never change what's on a server.  Blue/green deploys.  This help avoid "drift" through manual operations and long running infrastructure
+2, always change all servers, not a subset of them
+3, automation can fail, be prepared to put the old servers back exactly as they were
+
+Problems...
+
+1, Small changes are a ball ache (one line css) (moar automation)
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "ballache"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+2, debugging becomes harder as logs are shipped out, tools need to be reinstalled, cant fiddle with server setup
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Caps on "Debugging"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+3, storing data (not sure why he didnt suggest cloud but use containers for all the things, they are cheap and flexible, can be fired up from scratch)
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Caps on "Storing"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+Solutions...
+
+1, Automate all the things.  Operation s that have to happen can but they don't take any use time.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "Operations"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+Used to have a project called XL2web which converted Microsoft Excel spreadsheets and models into interactive Web applications, 
+Google basically asked them if they could do it in a browser so they built prototype in IE6 SP1 with an Excel emulator where the logic was all done on the server side.
+Google then acquired the company.
+They then added some persistence in the form or a database behind the state server.
+
+2007/08 was when Google started thinking about collaboration, they wanted to connect a second client to the same state.
+Technically, every mutation to the doc was stored in a log with snapshots generated at regular intervals so you could play back and forward mutations at will and quickly, this is when they put in the autosave functionality and "revisions" came "for free".
+
+
+However, when they got to character leve collboration (people editing the same word at the same time) they had to stop and think about what would be prctical.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "character level"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+
+Talked for a while about how they "solved" realtime collaboration.
+
+- mutations are applied at a revision number which dictates order, the order of mutations is critical to consistency across all clients.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Caps on first words of bullets. Full stops on bullets.

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+Use DNS in the same we we use t-bob to route traffic and give flexibility to what we server the client.
+
+
+### Culture
+
+Kev: best culture talk of the week, covered a lot of his use cases/problems and how you should look at them.  Essentially people need to think abou themselves and stop worrying about others.
+
+Reference to a book "It's your ship" by Abrashoff he stressed "make ownership bottom up".  The people at the top caring is not enough.
+"It's all about people" you model the culture and hope to see it playing back to you.
+His key point wsa "culture is event driven and organic, not mandated"
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: "was" 
"event-driven"

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+
+So where is culture defined? 
+- Real time, between people and groups
```
Comment (JonathanBD) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Caps and full stops.

**Comment by JonathanBD** [comment]
All done. Edits suggested.

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+title: QCon New York 2015 key takeaways - Part 2
+categories: conference
+author: kevin_hodges
+tags: conference qcon new york 2015
+comments: true
+---
+
+#### Introduction
+
+This is the second part of my notes and key take aways from Qonn New York 2015, a three day conference with main features (for me) on microservices, containers and culture.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Qonn

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+They can see how their architecture performs with an issue at any point.
+
+2. Who (scope)
+Scope starts on single request, single customer and increase slowly
+Can be limited to apps, regions etc to reduce risks
+
+3. Where (injection points)
+points that fail already or look like they might fail
+
+4. Failure itself
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: 5 w's, there are only 4 and this one is an F...

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+Each service/module implements a snippet to inject deplay where required.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: deplay?

File: _posts/2015-07-14-qcon-new-york-part-2.md (hunk 3 above)
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: 'button red top right'. Not sure this flows well

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+
+
+### Clojurescript demo
+#### Jearvon Darrie
+
+Kev: I'm afraid this was the only take away of any value.  A good one though right?
+
+"Programming is not about typing, it's about thinking" - Rich Hickey
+
+Immutable infrastructure with Docker and containers
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Should this have the h3 `###` like the other topics

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+- enable repeatable configuration
+
+Templates for services
+
+hardware - pre built when it comes in to the datacentre, they boot it up, smoke test and read to go (4hrs to get a server, apparently this is good :/)
+datacenters - "pre-baked" state is outside the machines, when it boots up it assumes a local configuration with time zone based management services
+containers
+- everything that isn't on the platform goes into a container, use openVZ, very stable, lightweight, simulate a full VM
+- each service is in its own container
+- then theres shared services nginx, squid, postfix, auth all in their own containers
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: there's

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+
+Use DNS in the same we we use t-bob to route traffic and give flexibility to what we server the client.
+
+
+### Culture
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: By whom?

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+
+Kev: best culture talk of the week, covered a lot of his use cases/problems and how you should look at them.  Essentially people need to think abou themselves and stop worrying about others.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: about

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+His key point wsa "culture is event driven and organic, not mandated"
+
+So where is culture defined? 
+- Real time, between people and groups
+- unexpected challenges
+- discovering what the rules/boundaries are
+- rituals that reinforce values
+- moments happen where people "step in", if there's an issue or a conversation and people take the lead in getting involved or directing
+
+Leaders set the culture people with titles. you dont have to be "titled" but be seen to lead when it's needed.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: This sentence doesn't flow very well

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+Culture cant be forced and if people are a detriment to that, you have to move them out or they will affect the culture.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: can't

**Comment by notABluesSinger** [comment]
Taking slot 2

File: _posts/2015-07-14-qcon-new-york-part-2.md [hunk 22]
```diff
@@ -0,0 +1,210 @@
...
+Templates for services
+
+hardware - pre built when it comes in to the datacentre, they boot it up, smoke test and read to go (4hrs to get a server, apparently this is good :/)
+datacenters - "pre-baked" state is outside the machines, when it boots up it assumes a local configuration with time zone based management services
+containers
+- everything that isn't on the platform goes into a container, use openVZ, very stable, lightweight, simulate a full VM
+- each service is in its own container
+- then theres shared services nginx, squid, postfix, auth all in their own containers
+
+Use DNS in the same we we use t-bob to route traffic and give flexibility to what we server the client.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: 'we we'? is this supposed ot be 'way we'

**Comment by notABluesSinger** [comment]
Perhaps this could do with a small summary section at the end... what your main takeaways are, how you think they could benefit ours and other businesses.

File: _posts/2015-07-14-qcon-new-york-part-2.md (hunk 14 above)
Comment (kevinhodges) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: Can't remember, must have lost track of what i was writing in my notes.  I'll remove the 5 w's header

File: _posts/2015-07-14-qcon-new-york-part-2.md (hunk 18 above)
Comment (kevinhodges) [inline _posts/2015-07-14-qcon-new-york-part-2.md]: no idea :)

**Comment by notABluesSinger** [comment]
:+1: from me

**Comment by JonathanBD** [comment]
Looks good to me.

**Pull request merged**